cng/rsa.go

```diff
@@ -318,7 +318,14 @@ func newPSS_PADDING_INFO(h crypto.Hash, saltLen int) (info bcrypt.PSS_PADDING_IN
 		return info, errors.New("crypto/rsa: unsupported hash function")
 	}
 	info.AlgId = utf16PtrFromString(hashID)
-	info.Salt = uint32(saltLen)
+	switch saltLen {
+	case -1: // rsa.PSSSaltLengthEqualsHash
+		info.Salt = uint32(h.Size())
+	case 0: // rsa.PSSSaltLengthAuto
+		err = errors.New("crypto/rsa: rsa.PSSSaltLengthAuto not supported")
```

Comment:
`ErrUnsupported` would be ideal, as it wouldn't require polluting the code with backend-specific checks. Unfortunately it would be an important performance penalty for the fallback cases, as sometimes we don't know a set of parameters are unsupported until we have already done several backend calls.

At the moment I'm planning to just do good-old backend-specific checks:

```go
if boring.Enabled &&
  (!goexperiment.CNGCrypto || (opts.saltLength() != PSSSaltLengthAuto && hash != crypto.MD5SHA1)) {

  bkey, err := boringPublicKey(pub)
  if err != nil {
    return err
  }
  if err := boring.VerifyRSAPSS(bkey, hash, digest, sig, opts.saltLength()); err != nil {
    return ErrVerification
  }
  return nil
}
```